spring-cloud-stream-binder-jms-activemq/src/main/java/org/springframework/cloud/stream/binder/jms/activemq/ActiveMQQueueProvisioner.java

```diff
@@ -65,26 +176,70 @@ private Queue createQueue(String topicName, Session session, String consumerName
 		return queue;
 	}
 
-	@Override
-	public String provisionDeadLetterQueue() {
-		Session session = null;
-		Connection connection = null;
-		try {
-			connection = connectionFactory.createConnection();
-			session = connection.createSession(true, 1);
-			session.createQueue(ACTIVE_MQ_DLQ);
-		} catch (JMSException e) {
-			e.printStackTrace();
-		}finally {
+	private final class JmsProducerDestination implements ProducerDestination {
+
+		private final Map<Integer, Topic> partitionTopics;
+
+		private JmsProducerDestination(Map<Integer, Topic> partitionTopics) {
+			this.partitionTopics = partitionTopics;
+		}
+
+		@Override
+		public String getProducerDestinationName() {
 			try {
-				JmsUtils.commitIfNecessary(session);
-				JmsUtils.closeSession(session);
-				JmsUtils.closeConnection(connection);
-			} catch (JMSException e) {
-				e.printStackTrace();
+				return partitionTopics.get(-1).getTopicName();
+			}
+			catch (JMSException e) {
+				if (logger.isInfoEnabled()) {
+					logger.info("JMS Exception", e);
+				}
+			}
+			return null;
+		}
+
+		@Override
+		public String getPartitionedProducerDestinationName(int partition) {
+			try {
+				return partitionTopics.get(partition).getTopicName();
+			}
+			catch (JMSException e) {
+				if (logger.isInfoEnabled()) {
+					logger.info("JMS Exception", e);
+				}
 			}
+			return null;
```

Comment:
Same here: `map.get()`.
No `JMSException`, but can be `NPE` if there is no entry for the provided `partition`.
What is our action then?
Is it really appropriate to return `null` to the caller?
Maybe some `IllegalArgumentException` would be better ?